src/toil/fileStores/nonCachingFileStore.py

```diff
@@ -101,6 +102,8 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=False,
 
         self.jobStore.readFile(fileStoreID, localFilePath, symlink=symlink)
         self.localFileMap[fileStoreID].append(localFilePath)
+        if fileStoreID.executable:
```

Comment:
This is one of the places where we need to handle the user sending us `str` instead of `FileID`. The `pack()`/`unpack()` stuff above is only used by CWL; Cactus isn't smart enough to use it (although it ought to be updated to).

If `fileStoreID` is a `str`, it won't have an `executable` attribute. You might want to replace this line with something that will supply a default value if the attribute is missing:

```
if getattr(fileStoreID, 'executable', False):
```